tests/components/uk_transport/test_sensor.py

```diff
@@ -16,85 +15,75 @@
-        """Test for operational uk_transport sensor with proper attributes."""
-        with requests_mock.Mocker() as mock_req, patch(
-            "homeassistant.util.dt.now", return_value=now().replace(hour=13)
-        ):
-            uri = re.compile(UkTransportSensor.TRANSPORT_API_URL_BASE + "*")
-            mock_req.get(uri, text=load_fixture("uk_transport_train.json"))
-            assert setup_component(self.hass, "sensor", {"sensor": self.config})
-            self.hass.block_till_done()
-
-        train_state = self.hass.states.get("sensor.next_train_to_WAT")
-
-        assert type(train_state.state) == str
-        assert train_state.name == f"Next train to {TRAIN_DESTINATION_NAME}"
-        assert train_state.attributes.get(ATTR_STATION_CODE) == TRAIN_STATION_CODE
-        assert train_state.attributes.get(ATTR_CALLING_AT) == TRAIN_DESTINATION_NAME
-        assert len(train_state.attributes.get(ATTR_NEXT_TRAINS)) == 25
-
-        assert (
-            train_state.attributes.get(ATTR_NEXT_TRAINS)[0]["destination_name"]
-            == "London Waterloo"
-        )
-        assert train_state.attributes.get(ATTR_NEXT_TRAINS)[0]["estimated"] == "06:13"
+async def test_bus(hass):
+    """Test for operational uk_transport sensor with proper attributes."""
+    with requests_mock.Mocker() as mock_req:
+        uri = re.compile(UkTransportSensor.TRANSPORT_API_URL_BASE + "*")
+        mock_req.get(uri, text=load_fixture("uk_transport_bus.json"))
+        assert await async_setup_component(hass, "sensor", VALID_CONFIG)
+        await hass.async_block_till_done()
+
+    bus_state = hass.states.get("sensor.next_bus_to_wantage")
+    assert None is not bus_state
+    assert f"Next bus to {BUS_DIRECTION}" == bus_state.name
+    assert BUS_ATCOCODE == bus_state.attributes[ATTR_ATCOCODE]
+    assert "Harwell Campus" == bus_state.attributes[ATTR_LOCALITY]
+    assert "Bus Station" == bus_state.attributes[ATTR_STOP_NAME]
+    assert 2 == len(bus_state.attributes.get(ATTR_NEXT_BUSES))
+
+    direction_re = re.compile(BUS_DIRECTION)
+    for bus in bus_state.attributes.get(ATTR_NEXT_BUSES):
+        assert None is not bus
+        assert None is not direction_re.search(bus["direction"])
+
+
+async def test_train(hass):
+    """Test for operational uk_transport sensor with proper attributes."""
+    with requests_mock.Mocker() as mock_req, patch(
+        "homeassistant.util.dt.now", return_value=now().replace(hour=13)
+    ):
+        uri = re.compile(UkTransportSensor.TRANSPORT_API_URL_BASE + "*")
+        mock_req.get(uri, text=load_fixture("uk_transport_train.json"))
+        assert await async_setup_component(hass, "sensor", VALID_CONFIG)
+        await hass.async_block_till_done()
+
+    train_state = hass.states.get("sensor.next_train_to_WAT")
+    assert None is not train_state
+    assert f"Next train to {TRAIN_DESTINATION_NAME}" == train_state.name
+    assert TRAIN_STATION_CODE == train_state.attributes[ATTR_STATION_CODE]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)